### PR TITLE
Avoid main context interference when running async tests.

### PR DESCRIPTION
**Pull request by tmiasko**
* Create separate context for each invocation instead of using global one.
* Configure thread default context before invoking closure so that
  operations will be executed in correct context.
* Avoid creating additional threads, they are unnecessary.

Fixes issue #93

### CODE REVIEW

**Comment by sdroege** [comment]
:+1:

File: src/test_util.rs [hunk 1]
```diff
@@ -4,25 +4,24 @@
...
 
     let (tx, rx) = channel();
 
-    thread::spawn(move || {
-        c.invoke(move || {
-            start(tx, l_clone);
-        });
+    c.invoke(move || {
+        let c = l_clone.get_context().unwrap();
+        c.push_thread_default();
```
Comment (sdroege) [inline src/test_util.rs]: I would do that simply at the top/bottom of the run_async function to have it for the whole runtime of the test

Comment (tmiasko) [reply src/test_util.rs]: Updated accordingly.

**Comment by sdroege** [comment]
@GuillaumeGomez Let's get this in then, together with Phil's PR this should solve most of the failures

**Comment by GuillaumeGomez** [comment]
Agreed. Just waiting for CIs. :)

**Pull request merged**